### PR TITLE
1780 product archive

### LINKED ISSUE #1780: unable to archive all options
With test product, archive option, publish, only one option is visible. archive this remaining second option, publish,  screen will display both archived options with one of them not having the archived label.  Clicking edit will also not open details in this state.
Server console:
```
Exception in queued task: Error: Could not find element with id SMr4rhDFnYvFMtDTX to change
    at SessionCollectionView.changed (packages/ddp-server/livedata_server.js:188:13)
    at Session.changed (packages/ddp-server/livedata_server.js:400:10)
    at Subscription.changed (packages/ddp-server/livedata_server.js:1254:19)
    at Object.removed (server/publications/collections/product.js:148:18)
```
Browser errors...
basic_reaction_product

### LINKED ISSUE #1799: [0.19.0] Tags undefined during product creation
Cannot add new product.

```
Exception in delivering result of invoking 'products/createProduct': ReferenceError: Tags is not defined
    at http://localhost:3000/app/app.js?hash=dcf5c5e77f0166f1b7b87bb5970dd98619079ee6:27006:22
    at MethodInvoker._callback (http://localhost:3000/packages/meteor.js?hash=e3f53db3be730057fed1a5f709ecd5fc7cae1229:1105:22)
    at MethodInvoker._maybeInvokeCallback (http://localhost:3000/packages/ddp-client.js?hash=bc32a166cd269e06a394f9418e0024d805bab379:3557:12)
    at MethodInvoker.dataVisible (http://localhost:3000/packages/ddp-client.js?hash=bc32a166cd269e06a394f9418e0024d805bab379:3586:10)
    at http://localhost:3000/packages/ddp-client.js?hash=bc32a166cd269e06a394f9418e0024d805bab379:4471:7
    at Array.forEach (native)
    at Function._.each._.forEach (http://localhost:3000/packages/underscore.js?hash=cde485f60699ff9aced3305f70189e39c665183c:149:11)
    at Connection._runAfterUpdateCallbacks (http://localhost:3000/packages/ddp-client.js?hash=bc32a166cd269e06a394f9418e0024d805bab379:4470:7)
    at Connection._performWrites (http://localhost:3000/packages/ddp-client.js?hash=bc32a166cd269e06a394f9418e0024d805bab379:4460:10)
    at Connection._flushBufferedWrites (http://localhost:3000/packages/ddp-client.js?hash=bc32a166cd269e06a394f9418e0024d805bab379:4423:10)

```

### PR DESCRIPTION
**Pull request by mikemurray**
Resolves #1780 
Resolves #1799 

Prevent instances where archiving a product / variant would also archive products
that are already archived, resulting in more revision documents being
created when not necessary.

**To Test**

0. From new product with top level and child variants
1. Archive product (very top level using the toolbar button)
2. Attempt to restore top level variant
3. Should receive an error that you cannot restore
4. Attempt to restore child variant
5. Should receive error that you cannot restore
6. Attempt to create a top level variant
7. Should receive error that you cannot create
8. Attempt to create a child variant
9. Should receive error that you cannot create

**tl;dr**

When a parent product or variant is deleted, you should not be able to add or restore the children because the dependency tree will be broken on publish.

### CODE REVIEW

**Comment by aaronjudd** [comment]
PR into master?

**Comment by aaronjudd** [comment]
Needs ii18n on alerts...

reaction_and_1780_product_archive_by_mikemurray_ _pull_request__1816_ _reactioncommerce_reaction

Also...

I archived, then published archived, then hit the undo, now archive button is disabled, product labeled "archived", publish is disabled. 
```
exception in delivering result of invoking 'products/deleteProduct': Error
    at http://localhost:3000/app/app.js?hash=1d80a06b422e88c031af5c817de19391a37c611f:59373:17
```
Attempting to clone:
```
    at Connection._flushBufferedWrites (http://localhost:3000/packages/ddp-client.js?hash=bc32a166cd269e06a394f9418e0024d805bab379:4423:10)
meteor.js?hash=e3f53db…:930 Exception in delivering result of invoking 'products/cloneProduct': Error
    at http://localhost:3000/app/app.js?hash=1d80a06b422e88c031af5c817de19391a37c611f:59310:13
```

File: imports/plugins/core/dashboard/client/containers/toolbarContainer.js [hunk 1]
```diff
@@ -2,7 +2,7 @@ import React from "react";
 import { composeWithTracker } from "/lib/api/compose";
```
Comment (aaronjudd) [inline imports/plugins/core/dashboard/client/containers/toolbarContainer.js]: Import order, should come after Meteor...

File: private/data/i18n/en.json [hunk 2]
```diff
@@ -302,9 +302,12 @@
         "variantDetails": "Variant Details",
         "duplicateVariant": "Duplicate",
         "addVariantOption": "Add Option",
+        "addVariantFail": "Unable to create variant for {{title}}. Check that it's parent product or variant is not archived.",
```
Comment (aaronjudd) [inline private/data/i18n/en.json]: These message could be reworded, and condensed.

File: imports/plugins/included/product-variant/client/templates/products/productDetail/variants/variantForm/variantForm.js [hunk 3]
```diff
@@ -167,7 +181,12 @@ Template.variantForm.events({
     }
     Meteor.call("products/cloneVariant", productId, template.data._id,
       function (error, result) {
-        if (result) {
+        if (error) {
+          Alerts.alert({
+            text: i18next.t("productDetailEdit.cloneVariantFail", { title: template.data.title }),
```
Comment (aaronjudd) [inline imports/plugins/included/product-variant/client/templates/products/productDetail/variants/variantForm/variantForm.js]: I suspect that `  "productDetailEdit": {` could all be moved into package `server/i18n`.
